code/mission/missiongoals.cpp

```diff
@@ -978,10 +978,11 @@ void mission_process_event( int event )
 	}
 
 	// decrement the trigger count.  When at 0, set the repeat count to 0 so we don't eval this function anymore
-	if (result && (Mission_events[event].trigger_count > 0) && (Mission_events[event].flags & MEF_USING_TRIGGER_COUNT) ) {
-		Mission_events[event].trigger_count--;
+	if (result && (Mission_events[event].trigger_count != 0) && (Mission_events[event].flags & MEF_USING_TRIGGER_COUNT) ) {
+		if (Mission_events[event].trigger_count > 0)
```

Comment:
Oh wow, whoops. I had an uncommitted change there that I thought I'd stashed; no idea how it accidentally wound up in this branch. Looks like it slipped in with bc0c7cf (the original vectorization commit), which was busy enough that I can imagine having accidentally included something unrelated.